src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -329,6 +330,108 @@ public bool IsNodeAddedRecently
             }
         }
 
+        /// <summary>
+        /// Gets a value indicating whether this model is frozen.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if this instance is frozen; otherwise, <c>false</c>.
+        /// </value>
+        public bool IsFrozen
+        {
+            get
+            {              
+                return NodeModel.IsFrozen;
+            }
+            set
+            {
+                NodeModel.IsFrozen = value;
```

Comment:
Per my comment above, if we need to raise a property change notification, I'd do it here to alert the UI, not in the model.